### ISSUE TITLE
Support for newer Python versions

### ISSUE THREAD

**Issue by bbudescu** [automl/SMAC3#1168]
At the time of this writing, the current latest stable CPython version is 3.13. However, the SMAC PyPI package only lists support for versions up to 3.10. For my and possibly others' use cases, it would be nice to be able to use SMAC with more recent versions for a few reasons like:
- support for newer language features, and not having to either give them up or maintain separate code versions
- performance gains in the target code for optimization (here, an alternative would be to run the target code with a different interpreter, which can be achieved by using the CLI)
- easier deployment on various OSs

Now, I can see there's a note in https://github.com/automl/SMAC3/issues/1116 about this, so I assume the problem is probably generated by compatibility with `pyrfr` via `swig` (that issue says "(see Eddies comment Feb 23rd in SMAC3)", but I couldn't find them :) ), so I guess the work is more complex, as I imagine it would probably entail either making, adding and maintaining some new pre-compiled binaries for `rfr`/`pyrfr`, or dropping it all together and replacing it, which in itself is a bag of snakes, because that's what benchmarking the performance of machine learning algorithm implementations usually is. I just wanted to leave this issue out here to show there's some interest for it. So no pressure, and thanks for your hard work :).

**Comment by benjamc** [comment]
Hi Bogdan,
thanks for your suggestion.
We are looking into supporting newer python versions.
I sometimes run SMAC in python 3.12 without any problems.
We are also in the process of replacing the random forest, however this might take a while until we properly benchmarked the new version.
Best

**Comment by berkgercek** [comment]
I'd like to follow up on this as Python 3.10 is reaching end-of-life [very soon](https://devguide.python.org/versions/) and will not longer receive security updates after October. Additionally 3.10 has not been receiving bugfixes since early 2023.

Are there any plans to support modern releases of Python in the near future?

**Comment by mwever** [comment]
Yes we are currently working on a new release that will allow us to support more modern versions of Python. The current bottleneck is depending on the pyrfr package. However, we worked on substituting the pyrfr package by scikit-learn and benchmarked SMAC already with this implementation to ensure its performance stays stable.

We are currently merging last features into our release branch before making the actual release public.
